default.nix

```diff
@@ -0,0 +1,222 @@
+{ pkgs ? import <nixpkgs> { }
+, stdenv ? pkgs.stdenv
+, lib ? pkgs.lib
+, ghostscript ? pkgs.ghostscript
+, ref ? null
+, doPdfGeneration ? true
+  # Checks whether the generated output matches the checked-in SSS32.ps.
+  # When doing development you may want to shut this off to obtain the
+  # output file that you need to check in.
+, doOutputDiff ? true
+}:
+
+let
+  src =
+    if isNull ref
+    then lib.sourceFilesBySuffices ./. [ ".ps" ".inc" ]
+    else
+      fetchGit {
+        url = ./.;
+        inherit ref;
+      };
+  shortId = lib.optionalString (! isNull ref) ("-" + builtins.substring 0 8 src.rev);
+
+
+  setup = rec {
```

Comment:
probably best to remove `rec`.

Reply:
I can remove it for now, but it will be used -- this `setup` set will contain all the various components of the postscript "setup" section. Each will have a `dependencies` array which refers to other setup components.